src/main/java/frc/robot/subsystems/drive/PhoenixDrive.java

```diff
@@ -0,0 +1,167 @@
+                        new ReplanningConfig(false, false)),
+                () -> DriverStation.getAlliance().orElse(Alliance.Blue) == Alliance.Red,
+                this);
+    }
+
+    private void startSimThread() {
+        lastSimTime = Utils.getCurrentTimeSeconds();
+        simNotifier =
+                new Notifier(
+                        () -> {
+                            final double currentTime = Utils.getCurrentTimeSeconds();
+                            double deltaTime = currentTime - lastSimTime;
+                            lastSimTime = currentTime;
+
+                            updateSimState(deltaTime, RobotController.getBatteryVoltage());
+                        });
+
+        simNotifier.startPeriodic(kSimLoopPeriod);
+    }
+
+    public ChassisSpeeds getCurrentSpeeds() {
+        return m_kinematics.toChassisSpeeds(getState().ModuleStates);
+    }
+
+    public void setGoalSpeeds(ChassisSpeeds goalSpeeds, boolean fieldCentric) {
+        System.out.println("attempt to apply");
+        SwerveRequest request;
+        if (fieldCentric) {
+            System.out.println("speed set to 1");
+            request =
+                    new SwerveRequest.FieldCentric()
+                            .withVelocityX(goalSpeeds.vxMetersPerSecond)
+                            .withVelocityY(goalSpeeds.vyMetersPerSecond)
+                            .withRotationalRate(goalSpeeds.omegaRadiansPerSecond)
+                            .withDeadband(0.0)
+                            .withRotationalDeadband(0.0)
+                            .withDriveRequestType(DriveRequestType.Velocity);
+        } else {
+            request =
+                    new SwerveRequest.RobotCentric()
+                            .withVelocityX(goalSpeeds.vxMetersPerSecond)
+                            .withVelocityY(goalSpeeds.vyMetersPerSecond)
+                            .withRotationalRate(goalSpeeds.omegaRadiansPerSecond)
+                            .withDeadband(0.0)
+                            .withRotationalDeadband(0.0)
+                            .withDriveRequestType(DriveRequestType.Velocity);
+        }
+        this.setControl(request);
+    }
+
+    @Override
+    public void simulationPeriodic() {
+        updateSimState(kSimLoopPeriod, RobotController.getBatteryVoltage());
+    }
+
+    @Override
+    public void periodic() {
+        if (!hasAppliedOperatorPerspective || DriverStation.isDisabled()) {
+            DriverStation.getAlliance()
+                    .ifPresent(
+                            (color) -> {
+                                this.setOperatorPerspectiveForward(
+                                        color == Alliance.Red
+                                                ? RedAlliancePerspectiveRotation
+                                                : BlueAlliancePerspectiveRotation);
+                            });
+            hasAppliedOperatorPerspective = true;
+        }
```

Comment:
how did we do this in 2024-Robot-Code?

Reply:
honzik used a conditional to flip values of joystick, i figured i would try out there way of doing it though to see if it led to performances in odometry and field positioning?